feature(eth): add eth_blobBaseFee

dzhalaevd opened the pull request:
### What was wrong?
Related to Issue #3527
Closes #3527

### How was it fixed?
- Support added for `eth_blobBaseFee`
- Async support added for `eth_blobBaseFee`
---
While running tests, the `TestEthereumTesterEthModule` raise that the RPC method eth_blobBaseFee is unknown. It seems like updating eth-tester might resolve the issue, but I am not entirely certain.

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md)

#### Cute Animal Picture

<img width="513" alt="image" src="https://github.com/user-attachments/assets/dff330f5-44d7-429e-b1c1-9e89182b6588">

dzhalaevd committed:
✨ add eth_blobBaseFee:

- 📝 add docs for eth_blobBaseFee
- 📝 add newsfragments
- 🧪 add test for eth_blobBaseFee

fselmo commented on docs/web3.eth.rst:
```diff
@@ -200,6 +200,18 @@ The following methods are available on the ``web3.eth`` namespace.
+    Fetches the expected base fee for blobs in the next block.
+    * Delegates to ``eth_blobBaseFee`` RPC Method
+
+    Returns the expected base fee in wei, represented as a hexadecimal.
```
> represented as a hexadecimal.

This may be true of the JSON-RPC spec, but the library returns the formatted value as an integer. Perhaps we can leave this simpler, borrowing from the `max_priority_fee_per_gas` description. We also need a blank line between the first sentence and the bullet (*) in order to render the docs properly.

>    Fetches the expected base fee for blobs in the next block.
> 
>    * Delegates to ``eth_blobBaseFee`` RPC Method
>
>    Returns the expected blob base fee in Wei.

This renders well

<img width="712" alt="Screenshot 2024-11-14 at 10 54 25" src="https://github.com/user-attachments/assets/74d950ed-7c77-4150-9fc4-087e9dd43347">

Thoughts?

dzhalaevd replied:
sounds reasonable. I'll fix it

fselmo reviewed (commented):
Hey @DavidRomanovizc. Thanks for the submission. I added a small note but I can also add tests here and address any issues with the tester provider. Are you OK with me picking this up and merging?

dzhalaevd commented:
> Hey @DavidRomanovizc. Thanks for the submission. I added a small note but I can also add tests here and address any issues with the tester provider. Are you OK with me picking this up and merging?

OK

fselmo committed:
Finish test setup for ``eth_blobBaseFee``:

- Add async test
- Add static value for the tester provider for now until support is added to eth-tester.

fselmo commented:
Looks good here. Thanks @DavidRomanovizc.
